Update Sonarr to v3

maxrogers opened the pull request:
# Test package for Sonarr V3 
### https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build 

⚠ To see which version you need: https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model 
⚠ Models that have `(x64)` listed on the wiki page should use the general x64 testing package.



### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

I just did a successful upgrade from Sonarr v2 to this package (v3) with no issues. It would be great to get a few experienced users to test and share their success/results going from v2 to v3 with this package. @Safihre can you make packages for people to test with? Thank you!

_Motivation:_  To add a v3 Sonarr package
_Reference:_  https://github.com/SynoCommunity/spksrc/pull/3518#issuecomment-552331259

Fringe09 commented:
I would like to test. Not sure how to access the package.

maxrogers commented:
@Fringe09, @Safihre is planning to compile for all of the packages and then post them here (he has a TravisCI pipeline set up.) As soon as he does that you can download and test.

Safihre commented:
So sorry I didn't see this last week! To the build machine! 🤖 🖥 

The releases can be downloaded here:
https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build
To see which version you need:
https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model (Models that have (x64) listed can use the general x64 testing packages)

I've only uploaded the packages, in case of problems please report them here so @maxrogers can look at them 👍

maxrogers commented:
Thanks @Safihre

Fringe09 commented:
I have never updated Sonarr via spk file before. Will this package detect v2x and upgrade to v3x automatically?

maxrogers commented:
@Fringe09, yes it should do just that. Complete a manual backup via Sonarr web-interface (a download a copy) before attempting (just in case.)

M3rofio commented:
hi @safihre
i have Synology DS3617xs -  synology_broadwell_3617xs DSM V 6.2.2 
installed Sonarr V2 want to update to v3 
i cant find the correct build https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build

maxrogers commented:
@M3rofio

`nzbdrone_x64-6.1_20191122-16.spk`

> https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model (Models that have (x64) listed can use the general x64 testing packages)

epheterson commented:
Wanted to chime in that I updated to v3 with the "nzbdrone_x64-6.1_20191122-16.spk" package on my Synology DS415+ and everything went smooth as butter! Thanks so much :)

ymartin59 commented on spk/sonarr/Makefile:
```diff
@@ -18,7 +18,7 @@ DESCRIPTION_SPN = Sonarr es un PVR para los usuarios de grupos de noticias y tor
 DISPLAY_NAME = Sonarr
 STARTABLE = yes
-CHANGELOG = "Update Sonarr to v2.0.0.5301."
+CHANGELOG = "Update Sonarr to v3"
```
```suggestion
CHANGELOG = "Update Sonarr to v3.0.3.654"
```

ymartin59 commented on spk/sonarr/Makefile:
```diff
@@ -1,6 +1,6 @@
 SPK_NAME = nzbdrone
 SPK_VERS = $(shell date +%Y%m%d)
```
I would really rename package too and set its upstream version...
I guess it should be possible to automate transition, copying `nzbdrone/target/var` to new `sonarr/target/var` and triggering `nzbdrone` package removal.
@maxrogers May you please investigate such package transition? It would be really helpful for other applications too.

Safihre replied:
Can we trigger an official package removal from within the script? Such that it's also removed from the the package manager?

ymartin59 replied:
I think so. `synopkg uninstall` but must be tested in case it cannot be triggered when installation is already in progress. https://forum.synology.com/enu/viewtopic.php?t=144513

maxrogers replied:
https://github.com/SynoCommunity/spksrc/pull/3803#issuecomment-564166738

TimeBomb commented:
Wanted to mention I used ` nzbdrone_x64-6.1_20191122-16.spk` to upgrade Sonarr on my Synology DS1019+ and it worked perfectly. All preexisting content is still visible, and downloading new content works as expected. Thanks!

maxrogers commented:
@ymartin59 Yes, I will look into this as soon as I have some free time.

maxrogers commented:
@ymartin59 I have made a package that successfully upgrades while also renaming from nzbdrone to sonarr. However, the method is too crude and I do not think it is worth doing it just to change the internal naming of the package (from `nzbdrone`). It opens some potential install issues and I do not think is wise to do for a change that is transparent to the users (only to us).

For example:
It seems that the installer does not allow enough privilege to stop or uninstall via the `synopkg` utility. 

Here are the errors that I get when trying both:

**synopkg stop nzbdrone**
`Failed to stop package 275`
**synopkg uninstall nzbdrone**
`Failed to uninstall package: nzbdrone, 275`

I tried adding `sudo` with no luck. I assume synology does not allow the utility to be used during an install... Both commands work great via command line (as root)

Because the installer does not allow us to use synopkg during an install we have to manually kill processes and then manually delete the package from `/var/packages/` and `@appstore`. 

Additionally, the package installs as a new install (not an upgrade) therefore to keep the same port number we have to install the package with a different port number and then modify three files to use the correct port number. In addition, there are a few other issues that require adding more complexity service-setup.sh to address the issue of performing upgrade tasks in the install portion due to the package installing as a new install (not an upgrade).

NOTE: I also attempted to do the package name change via an upgrade of the nzbdrone package. However, that method introduces potential user permission issues and will also require a second follow-up package to fully rename the package to `sonarr` from `nzbdrone`. With a second package, we have no method to ensure that the users install the first before the second...

Idus commented:
Stumbled upon this, since I had issues with v2. So, just did the update on my DS918+ with the "generic x64"-package - worked like a charm! Asked for an extra minor update - and then just got to work. Thanx!

While I am at it - am I missing that there is a **_Radarr v.3_** lurking on github somewhere as well? Could not find any when I searched?

Kudos to all you guys who have the skills (and time) to do this!

Safihre commented:
@maxrogers Thanks for looking into this. But now how to proceed? Seems we have to wait for Sonarr to officially stop with V2 before we can push this package, since it will replace the V2 package.
Maybe we can ask the Sonarr (@Taloth/@markus101) to link from their downloads-page (https://sonarr.tv/#downloads-v3-nas) to this PR for users that want to install the test-packages? I updated the starting post with big instructions.

maxrogers commented:
@Safihre I think that sounds like the best plan. 

While v3 is still considered beta [sonarr.tv](https://sonarr.tv/#downloads-v3-nas) can link to the v3 packages you have posted. As soon as v3 is no longer considered beta, then we can publish to the repo.

That all seems very reasonable to me. Thanks!

Adelscott commented:
Thanks a lot guys, was struggling with V2 and language filter problems. Found this. Migration worked like a charm on DS713+ !

Fringe09 commented:
Can the packages @ https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build be also used for clean installs?

maxrogers commented:
> Can the packages @ https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build be also used for clean installs?

Yes

Azzaro78 commented:
Would it be possible to package the latest build? Mine is stuck with the tasks being stuck and update internally fails. 

Not sure how else I can get the later version installed (3.0.3.721)

or if there are some easy steps to follow to install the latest version manually, I'd be happy to try.

Fringe09 commented:
I faced the exact same issue. I took a backup of the db, uninstalled, installed from scratch, updated to the latest version, and then restored the db.

Back and restore instruction are located @ https://github.com/Sonarr/Sonarr/wiki/Backup-and-Restore

M0UL commented:
it should be considered releasing the v3 package with [thetvdb issues ](https://old.reddit.com/r/sonarr/comments/fhj8z9/important_zero_episodes_and_old_episodes/l) going on. and that fixes are only suplied for v3.

rp4k commented:
> @Safihre I think that sounds like the best plan.
> 
> While v3 is still considered beta [sonarr.tv](https://sonarr.tv/#downloads-v3-nas) can link to the v3 packages you have posted. As soon as v3 is no longer considered beta, then we can publish to the repo.
> 
> That all seems very reasonable to me. Thanks!

do we know  when this will be happening. because syno still shows v2

Safihre commented:
@rp4k best place to ask is at the Sonarr forums! In the meantime you can use this package.

bgalakazam commented:
I am glad I found this page. Wanted to get some of the extra functionality of v3. My little NAS that could, DS118, is still chugging along. Keep up the good work and thank you.

epheterson commented:
Slightly off topic, but Sonarr v3 is so great... Does anybody here know if there's a Radarr v3 package for Synology available (or is anybody here willing to make one for testing)?

epheterson commented:
BTW this may be a dupe of: https://github.com/SynoCommunity/spksrc/issues/3443

Safihre commented:
Did you even read the [original post](https://github.com/SynoCommunity/spksrc/pull/3803#issue-340111216) here?

epheterson commented:
> Did you even read the [original post](https://github.com/SynoCommunity/spksrc/pull/3803#issue-340111216) here?

Yeah whoops didn't realize this was the pull request and that was an issue ticket (at a glance thought both were issues), my bad!

M0UL commented:
is possible to get new test build? since the current  v2 package keeps asking for a update.

maxrogers commented:
> is possible to get a new test build? since the current v2 package keeps asking for a update.

No need, the update you are being prompted for has v2 & v3 compatibility since this: https://github.com/SynoCommunity/spksrc/commit/117650e6eee4b3186507d1011f28ff94cf6d58bb#diff-fba6cd69807057b16bb679cfdc048ce5

I just updated just fine while running v3. Make a backup via web-ui (just in case)

M0UL commented:
> 
> 
> > is possible to get a new test build? since the current v2 package keeps asking for a update.
> 
> No need, the update you are being prompted for has v2 & v3 compatibility since this: [117650e#diff-fba6cd69807057b16bb679cfdc048ce5](https://github.com/SynoCommunity/spksrc/commit/117650e6eee4b3186507d1011f28ff94cf6d58bb#diff-fba6cd69807057b16bb679cfdc048ce5)
> 
> I just updated just fine while running v3. Make a backup via web-ui (just in case)
cool. thanks for me letting me know

epheterson commented:
This file is the same as the current v3 with the version number bumped by 1 so Synology doesn't suggest updating (also fixes the setup if you managed to install that update, like I did).

[nzbdrone_x64-6.1_20200922-16.spk.zip](https://github.com/SynoCommunity/spksrc/files/5362619/nzbdrone_x64-6.1_20200922-16.spk.zip)

schumi2004 commented:
Would it be possible to generate new packages with version numbers bumped so people can upgrade existing version instead of playing around with backups?

maxrogers commented:
For clarification, if you are already running Sonarr v3 just update when prompted using the package center (synocommunity). The current package available via package center is the latest and it supports running either v2 or v3. If you want the latest package just update as prompted it is v3 compatible and will run either v2 or v3.

Previously Sonarr did not offer the ability to switch to v3 from a v2 web ui (switch branch). Otherwise that would be the easiest way to upgrade. I assume that is still the case but I have not had the time to check. For now folks will need to use the above packages to transition and then just upgrade per normal via package center updates.

schumi2004 commented:
@maxrogers 
Currently I’m still running Sonarr V2, v2.0.0.5344 to be precise, and SynoCommunity package version 20200921-18 and this means I cannot use above package because it has a older revision number.
If I understand the Wiki correctly you also cannot switch between version branches v2 w/ v3 and vice-versa.
I don't want to start from scratch so please advise the best approach to upgrade to v3 without losing data? 
I know the hard way but prefer a somewhat easier way 😉

Safihre commented:
I have rebased the branch to `master` and am now building new packages. They should appear soon.

th0ma7 commented:
> I have rebased the branch to `master` and am now building new packages. They should appear soon.

I would suggest rebasing again so you have updated code from PR #4219 for package generation.

Stanzilla commented:
Hey @Safihre any chance we could get an updated package with the latest workarounds for mono problems included?

Safihre commented:
I rebased and started a new build.

Stanzilla commented:
Nice, might be worth manually adding the workaround to 88f6282 until @hgy59 had time to add it.

voltrongreen commented:
Hello
I can't unerstand what version I need to download for Synology 214+, help me pls

hgy59 commented:
> 
> 
> Hello
> I can't unerstand what version I need to download for Synology 214+, help me pls

On the wiki page [Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures) you can see that for DS214+ with armadaxp you need the package for armv7 (generic) arch.

voltrongreen commented:
> > Hello
> > I can't unerstand what version I need to download for Synology 214+, help me pls
> 
> On the wiki page [Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures) you can see that for DS214+ with armadaxp you need the package for armv7 (generic) arch.

None of the above say armv7, I already tried 3 with no luck.

supergroover commented:
> > > Hello
> > > I can't unerstand what version I need to download for Synology 214+, help me pls
> > 
> > 
> > On the wiki page [Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures) you can see that for DS214+ with armadaxp you need the package for armv7 (generic) arch.
> 
> None of the above say armv7, I already tried 3 with no luck.

I can't do it either, I have a synology ds414 with an armadaxp.
i've tried:
- nzbdrone_armada370-6.1_20201116-19.spk
- nzbdrone_armada375-6.1_20201116-19.spk
- nzbdrone_armada37xx-6.2_20201116-19.spk
- nzbdrone_armada38x-6.1_20201116-19.spk

schumi2004 commented:
> > > > Hello
> > > > I can't unerstand what version I need to download for Synology 214+, help me pls
> > > 
> > > 
> > > On the wiki page [Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures) you can see that for DS214+ with armadaxp you need the package for armv7 (generic) arch.
> > 
> > 
> > None of the above say armv7, I already tried 3 with no luck.
> 
> I can't do it either, I have a synology ds414 with an armadaxp.
> i've tried:
> 
> * nzbdrone_armada370-6.1_20201116-19.spk
> * nzbdrone_armada375-6.1_20201116-19.spk
> * nzbdrone_armada37xx-6.2_20201116-19.spk
> * nzbdrone_armada38x-6.1_20201116-19.spk

Can you download this package and try the armv7 version?
https://github.com/SynoCommunity/spksrc/suites/1638874483/artifacts/30315924

supergroover commented:
this package works, too bad it's Radarr and not Sonarr 😀


Op vr 11 dec. 2020 om 10:49 schreef schumi2004 <notifications@github.com>:

> Hello
> I can't unerstand what version I need to download for Synology 214+, help
> me pls
>
> On the wiki page Package Architectures
> <https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures>
> you can see that for DS214+ with armadaxp you need the package for armv7
> (generic) arch.
>
> None of the above say armv7, I already tried 3 with no luck.
>
> I can't do it either, I have a synology ds414 with an armadaxp.
> i've tried:
>
>    - nzbdrone_armada370-6.1_20201116-19.spk
>    - nzbdrone_armada375-6.1_20201116-19.spk
>    - nzbdrone_armada37xx-6.2_20201116-19.spk
>    - nzbdrone_armada38x-6.1_20201116-19.spk
>
> Can you download this package and try the armv7 version?
>
> https://github.com/SynoCommunity/spksrc/suites/1638874483/artifacts/30315924
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynoCommunity/spksrc/pull/3803#issuecomment-743092879>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQBO5EKHTUSPRPQJGPDDU3SUHTLDANCNFSM4JMJAV5Q>
> .
>

cfpsystems commented:
Hi, I read through this and didn't see the same situation. I am running a DSM VM of DSM7 on a DS918+. I downloaded the x64 package per the processor page and when I manually attempt to install it, it tells me "unable to install sonar because it runs with root privleges...Please contact the developer to change the privleges of the package"
Any ideas?
Thanks!

tgcentral commented:
> > > > Hello
> > > > I can't unerstand what version I need to download for Synology 214+, help me pls
> > > 
> > > 
> > > On the wiki page [Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures) you can see that for DS214+ with armadaxp you need the package for armv7 (generic) arch.
> > 
> > 
> > None of the above say armv7, I already tried 3 with no luck.
> 
> I can't do it either, I have a synology ds414 with an armadaxp.
> i've tried:
> 
> * nzbdrone_armada370-6.1_20201116-19.spk
> * nzbdrone_armada375-6.1_20201116-19.spk
> * nzbdrone_armada37xx-6.2_20201116-19.spk
> * nzbdrone_armada38x-6.1_20201116-19.spk

I have a DS214+ (armadaxp) so also ran into the same problem. I downloaded the source code tarball under [sonarr-v3-build](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build), and built an spk. I upgraded my own Sonarr installation from v2 to v3, and then tested again by uninstalling completely and upgrading from a fresh v2 install.

Happy to share the spk for other armadaxp owners (make sure you take Sonarr backups beforehand!) but am a Github noob so am not sure what the correct/best practice way is for sharing. Can anyone advise?

supergroover commented:
> > > > > Hello
> > > > > I can't unerstand what version I need to download for Synology 214+, help me pls
> > > > 
> > > > 
> > > > On the wiki page [Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures) you can see that for DS214+ with armadaxp you need the package for armv7 (generic) arch.
> > > 
> > > 
> > > None of the above say armv7, I already tried 3 with no luck.
> > 
> > 
> > I can't do it either, I have a synology ds414 with an armadaxp.
> > i've tried:
> > 
> > * nzbdrone_armada370-6.1_20201116-19.spk
> > * nzbdrone_armada375-6.1_20201116-19.spk
> > * nzbdrone_armada37xx-6.2_20201116-19.spk
> > * nzbdrone_armada38x-6.1_20201116-19.spk
> 
> I have a DS214+ (armadaxp) so also ran into the same problem. I downloaded the source code tarball under [sonarr-v3-build](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build), and built an spk. I upgraded my own Sonarr installation from v2 to v3, and then tested again by uninstalling completely and upgrading from a fresh v2 install.
> 
> Happy to share the spk for other armadaxp owners (make sure you take Sonarr backups beforehand!) but am a Github noob so am not sure what the correct/best practice way is for sharing. Can anyone advise?

Maybe you can upload the file to https://mega.nz/ and share the link with armadaxp owners.

tgcentral commented:
> > I have a DS214+ (armadaxp) so also ran into the same problem. I downloaded the source code tarball under [sonarr-v3-build](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build), and built an spk. I upgraded my own Sonarr installation from v2 to v3, and then tested again by uninstalling completely and upgrading from a fresh v2 install.
> > Happy to share the spk for other armadaxp owners (make sure you take Sonarr backups beforehand!) but am a Github noob so am not sure what the correct/best practice way is for sharing. Can anyone advise?
> 
> Maybe you can upload the file to https://mega.nz/ and share the link here.

Sure, here it is - [link](https://mega.nz/file/6J0UkIaC#S2P7KlV3tJfD5jBrpBDZxBrTWWkwMzgiC_EZ7Vy4nWo)

supergroover commented:
> > > I have a DS214+ (armadaxp) so also ran into the same problem. I downloaded the source code tarball under [sonarr-v3-build](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build), and built an spk. I upgraded my own Sonarr installation from v2 to v3, and then tested again by uninstalling completely and upgrading from a fresh v2 install.
> > > Happy to share the spk for other armadaxp owners (make sure you take Sonarr backups beforehand!) but am a Github noob so am not sure what the correct/best practice way is for sharing. Can anyone advise?
> > 
> > 
> > Maybe you can upload the file to https://mega.nz/ and share the link here.
> 
> Sure - [link](https://mega.nz/file/6J0UkIaC#S2P7KlV3tJfD5jBrpBDZxBrTWWkwMzgiC_EZ7Vy4nWo)

Great, this package is working. Thank you!

ccwillia commented:
I downloaded nzbdrone_88f6281-6.1_20201116-19.spk for my ds211j and the service will not start

mono_gdb_render_native_backtraces not supported on this platform, unable to find gdb or lldb

ccwillia commented:
> I downloaded nzbdrone_88f6281-6.1_20201116-19.spk for my ds211j and the service will not start
> 
> mono_gdb_render_native_backtraces not supported on this platform, unable to find gdb or lldb
> 
> I was able to resolve this by modifying service-setup and replacing 88f8621 with 88f6281. Looking at the [architecture page ](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model) this workaround includes the wrong architecture versions they should be 88f6281 and 88f6282 instead what shown below 88f8621 and 88f8622
> 
> ```
> # workaround for mono bug with armv5 (https://github.com/mono/mono/issues/12537)
> if [ "$SYNOPKG_DSM_ARCH" == "88f8621" -o "$SYNOPKG_DSM_ARCH" == "88f8622" ]; then
>     MONO="MONO_ENV_OPTIONS='-O=-aot,-float32' ${MONO_PATH}/mono"
> fi
> ```

Even though the service starts there are lots of errors in the logs when clicking things like fetch updates

ccwillia commented:
@Safihre Can you rebuild the nzbdrone_88f6281-6.1_20201116-19.spk package list [here](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build) with the most current code based

Safihre commented:
I'm sorry I can't, Travis has limited the number of builds and I have used all my build minutes.

hgy59 commented:
> 
> 
> @Safihre Can you rebuild the nzbdrone_88f6281-6.1_20201116-19.spk package listed [here](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build) with the most current code based

@ccwillia you can download the latests build of the github build action
- click details on one of the build tasks under `All checks have passed`
- Open the popup menu under `Artifacts`
- Click `packages`

This will download `packages.zip` containing all successfully built packages.

Extract the package for your arch (currently: `nzbdrone_88f6281-6.1_20210126-19.spk`).

repairmant commented:
DS1817+ | Intel Atom C2538 | Avoton | avoton | x86_64 is my unit and running on DSM 6.2.x not sure what package I should get. Any ideas? or is there a updating coming out to release to SynoCommunity in the package manager very soon cause mine it totally not functional until I get the v3 update either beta or release.

ccwillia commented:
> > @Safihre Can you rebuild the nzbdrone_88f6281-6.1_20201116-19.spk package listed [here](https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build) with the most current code based
> 
> @ccwillia you can download the latest build of the github build action
> 
> * click details on one of the build tasks under `All checks have passed`
> * Open the popup menu under `Artifacts`
> * Click `packages`
> 
> This will download `packages.zip` containing all successfully built packages.
> 
> Extract the package for your arch (currently: `nzbdrone_88f6281-6.1_20210126-19.spk`).

I extracted and installed the package above.  Sonarr V3 is able to start but appears to have issues accessing the internet.  I have tried update as well as adding an indexer, both give a similar error `System.DllNotFoundException`

```
[Warn] Newznab: Unable to connect to indexer 

[v3.0.4.1091] System.Net.WebException: Error: SecureChannelFailure (Authentication failed, see inner exception.): 'https://api.nzb.su/api?t=caps&apikey=****' ---> System.Net.WebException: Error: SecureChannelFailure (Authentication failed, see inner exception.) ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> System.DllNotFoundException: /volume1/@appstore/mono/lib/../lib/libmono-btls-shared.so
  at (wrapper managed-to-native) Mono.Btls.MonoBtlsSslCtx.mono_btls_ssl_ctx_new()
  at Mono.Btls.MonoBtlsSslCtx..ctor () [0x00000] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Btls.MonoBtlsContext.InitializeConnection () [0x00000] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Btls.MonoBtlsContext.StartHandshake () [0x00000] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) [0x0007e] in <c5f4e317158343a5916a953327f92540>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
  at Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) [0x00006] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) [0x000ff] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.AsyncProtocolRequest.StartOperation (System.Threading.CancellationToken cancellationToken) [0x0008b] in <c5f4e317158343a5916a953327f92540>:0 
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessAuthentication (System.Boolean runSynchronously, Mono.Net.Security.MonoSslAuthenticationOptions options, System.Threading.CancellationToken cancellationToken) [0x00252] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.MonoTlsStream.CreateStream (System.Net.WebConnectionTunnel tunnel, System.Threading.CancellationToken cancellationToken) [0x00126] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebConnection.CreateStream (System.Net.WebOperation operation, System.Boolean reused, System.Threading.CancellationToken cancellationToken) [0x001ba] in <c5f4e317158343a5916a953327f92540>:0 
   --- End of inner exception stack trace ---
  at System.Net.WebConnection.CreateStream (System.Net.WebOperation operation, System.Boolean reused, System.Threading.CancellationToken cancellationToken) [0x0021a] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebConnection.InitConnection (System.Net.WebOperation operation, System.Threading.CancellationToken cancellationToken) [0x00141] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebOperation.Run () [0x0009a] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebCompletionSource`1[T].WaitForCompletion () [0x00094] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.HttpWebRequest.GetResponse () [0x00016] in <c5f4e317158343a5916a953327f92540>:0 
  at NzbDrone.Common.Http.Dispatchers.ManagedHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x00123] in <ada5f4b7b39e40ea884a80b652a9fe4b>:0 
   --- End of inner exception stack trace ---
  at NzbDrone.Common.Http.Dispatchers.ManagedHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x001e7] in <ada5f4b7b39e40ea884a80b652a9fe4b>:0 
  at NzbDrone.Common.Http.HttpClient.ExecuteRequest (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookieContainer) [0x00080] in <ada5f4b7b39e40ea884a80b652a9fe4b>:0 
  at NzbDrone.Common.Http.HttpClient.Execute (NzbDrone.Common.Http.HttpRequest request) [0x000b0] in <ada5f4b7b39e40ea884a80b652a9fe4b>:0 
  at NzbDrone.Common.Http.HttpClient.Get (NzbDrone.Common.Http.HttpRequest request) [0x00007] in <ada5f4b7b39e40ea884a80b652a9fe4b>:0 
  at NzbDrone.Core.Indexers.Newznab.NewznabCapabilitiesProvider.FetchCapabilities (NzbDrone.Core.Indexers.Newznab.NewznabSettings indexerSettings) [0x000a1] in <174bb22c859448a290bba5941b4f68d1>:0 
  at NzbDrone.Core.Indexers.Newznab.NewznabCapabilitiesProvider+<>c__DisplayClass4_0.<GetCapabilities>b__0 () [0x00000] in <174bb22c859448a290bba5941b4f68d1>:0 
  at NzbDrone.Common.Cache.Cached`1[T].Get (System.String key, System.Func`1[TResult] function, System.Nullable`1[T] lifeTime) [0x000b1] in <ada5f4b7b39e40ea884a80b652a9fe4b>:0 
  at NzbDrone.Core.Indexers.Newznab.NewznabCapabilitiesProvider.GetCapabilities (NzbDrone.Core.Indexers.Newznab.NewznabSettings indexerSettings) [0x00046] in <174bb22c859448a290bba5941b4f68d1>:0 
  at NzbDrone.Core.Indexers.Newznab.Newznab.get_PageSize () [0x0000c] in <174bb22c859448a290bba5941b4f68d1>:0 
  at NzbDrone.Core.Indexers.Newznab.Newznab.GetRequestGenerator () [0x0000b] in <174bb22c859448a290bba5941b4f68d1>:0 
  at NzbDrone.Core.Indexers.HttpIndexerBase`1[TSettings].TestConnection () [0x00007] in <174bb22c859448a290bba5941b4f68d1>:0 
```

Stanzilla commented:
Yeah you can't use it on 88f6281, the mono version does not work.

ccwillia commented:
> Yeah you can't use it on 88f6281, the mono version does not work.

So there i not a version of mono which works with Sonarr V3 on 88f6281?

Stanzilla commented:
> > Yeah you can't use it on 88f6281, the mono version does not work.
> 
> So there i not a version of mono which works with Sonarr V3 on 88f6281?

Correct!

Taloth commented:
@Safihre v3 is being released upstream. Also added another endpoint for clearer release channel info https://services.sonarr.tv/v1/releases
Ping me on discord if you need details.

_Last 2 months have been long hours for me to migrate our backend service to new hardware. So I wanted to wait till that was done._

maxrogers commented:
> @Safihre v3 is being released upstream. Also added another endpoint for clearer release channel info https://services.sonarr.tv/v1/releases
> Ping me on discord if you need details.
> 
> _Last 2 months have been long hours for me to migrate our backend service to new hardware. So I wanted to wait till that was done._

Awesome! Thanks @Taloth 

@Safihre I updated/squashed this PR to use the newly released v3-stable and new release channel

Wuszek commented:
Is this a mono version? Or using .NET core like new Radarr update from @ta264?

ta264 commented:
This is mono.  Sonarr doesn't yet support .NET core.

Safihre commented:
I can not do anything more, someone else should merge and release this like @hgy59 or @publicarray !

Wuszek commented:
> This is mono. Sonarr doesn't yet support .NET core.

Meh :P

Wuszek commented:
I can confirm, that this package works and I upgraded from v2 to v3 without any issues. 

Besides, the only MONO package working with Jackett and Sonarr is archaic 5.8... All newer versions have SSL issue, that won't let Jackerr/Sonarr work properly.

Taloth commented:
> All newer versions have SSL issue, that won't let Jackerr/Sonarr work properly.

That'd be specific to the syno mono package i presume. I've heard some issues with realtek cpus and elliptic curve issue in btls but that's about it.
The issue posted earlier in this PR mentions a missing library, that would _probably_ be a packaging issue. Someone should investigate if the mentioned library exists.

Note that Sonarr v3 will display a persistent Health Check notice for versions below 5.18. I really want to bump that to 'not support anymore' over the course of the v3 lifecycle. Since anything below 5.18 prevents us from apply some necessary updates to dependencies.

Wuszek commented:
I think it is specific to syno mono package, but it has not been resolved since nearly 2 years now..

Stanzilla commented:
Yeah that's not really related to this PR/Sonarr but it looks like it's the end of the line for mono based software on 88f6281.

publicarray commented:
@maxrogers

The first time I did the migration the database wasn't migrated (not sure if it was a fluke). The following times the database migrated every time.

I would prefer to reduce the number of errors on installation. Maybe some checking if the folder exists?

<details>
<summary>Install log</summary>
<pre>
Thu Mar 11 13:43:05 AEDT 2021
===> Step preinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 13:43:06 AEDT 2021
===> Step postinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Installing service configuration /var/packages/nzbdrone/conf/nzbdrone.sc
Adding 'sc-nzbdrone' to 'sc-download'
Group Name: [sc-download]
Group Type: [AUTH_LOCAL]
Group ID:   [65538]
Group Members: 
0:[sc-lidarr]
1:[sc-radarr]
2:[sc-nzbdrone]
Invoke service_postinst
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var/.config
chown: cannot access '/volume1/@appstore/nzbdrone/var/.config': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var
Thu Mar 11 15:11:35 AEDT 2021
===> Step preupgrade. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Invoke service_preupgrade
mkdir: cannot create directory '/volume1/@appstore/nzbdrone/var/.config': File exists
Installed Sonarr Binary: 2.0.0.5344
Requested Sonarr Binary: 3.0.5.1144
[REPLACING] Installed Sonarr Binary
Thu Mar 11 15:11:36 AEDT 2021
===> Step preuninst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:11:36 AEDT 2021
===> Step postuninst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:11:36 AEDT 2021
===> Step preinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:11:36 AEDT 2021
===> Step postinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Installing service configuration /var/packages/nzbdrone/conf/nzbdrone.sc
Invoke service_postinst
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var/.config
chown: cannot access '/volume1/@appstore/nzbdrone/var/.config': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var
Thu Mar 11 15:11:36 AEDT 2021
===> Step postupgrade. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var
Invoke service_postupgrade
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var/.config
Granting 'sc-nzbdrone' unix ownership on /tmp/nzbdrone_backup
chown: cannot access '/tmp/nzbdrone_backup': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /tmp/nzbdrone_update
chown: cannot access '/tmp/nzbdrone_update': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /tmp/sonarr_backup
chown: cannot access '/tmp/sonarr_backup': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /tmp/sonarr_update
chown: cannot access '/tmp/sonarr_update': No such file or directory
Thu Mar 11 15:14:01 AEDT 2021
===> Step preuninst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Removing service configuration nzbdrone.sc
Thu Mar 11 15:14:01 AEDT 2021
===> Step postuninst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Removing user sc-nzbdrone
Thu Mar 11 15:17:40 AEDT 2021
===> Step preinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:17:40 AEDT 2021
===> Step postinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Installing service configuration /var/packages/nzbdrone/conf/nzbdrone.sc
Adding 'sc-nzbdrone' to 'sc-download'
Group Name: [sc-download]
Group Type: [AUTH_LOCAL]
Group ID:   [65538]
Group Members: 
0:[sc-lidarr]
1:[sc-radarr]
2:[sc-nzbdrone]
Invoke service_postinst
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var/.config
chown: cannot access '/volume1/@appstore/nzbdrone/var/.config': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var
Thu Mar 11 15:19:37 AEDT 2021
===> Step preupgrade. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Invoke service_preupgrade
mkdir: cannot create directory '/volume1/@appstore/nzbdrone/var/.config': File exists
Installed Sonarr Binary: 2.0.0.5344
Requested Sonarr Binary: 3.0.5.1144
[REPLACING] Installed Sonarr Binary
Thu Mar 11 15:19:37 AEDT 2021
===> Step preuninst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:19:37 AEDT 2021
===> Step postuninst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:19:37 AEDT 2021
===> Step preinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Thu Mar 11 15:19:38 AEDT 2021
===> Step postinst. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Installing service configuration /var/packages/nzbdrone/conf/nzbdrone.sc
Invoke service_postinst
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var/.config
chown: cannot access '/volume1/@appstore/nzbdrone/var/.config': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var
Thu Mar 11 15:19:38 AEDT 2021
===> Step postupgrade. USER=nzbdrone GROUP=sc-download SHARE_PATH=
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var
Invoke service_postupgrade
Granting 'sc-nzbdrone' unix ownership on /volume1/@appstore/nzbdrone/var/.config
Granting 'sc-nzbdrone' unix ownership on /tmp/nzbdrone_backup
chown: cannot access '/tmp/nzbdrone_backup': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /tmp/nzbdrone_update
chown: cannot access '/tmp/nzbdrone_update': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /tmp/sonarr_backup
chown: cannot access '/tmp/sonarr_backup': No such file or directory
Granting 'sc-nzbdrone' unix ownership on /tmp/sonarr_update
chown: cannot access '/tmp/sonarr_update': No such file or directory

</pre>
</details>

publicarray commented on spk/sonarr/src/service-setup.sh:
```diff
@@ -90,17 +82,21 @@ service_postupgrade ()
     # new updates/backups will fail due to permissions (see #3185)
-    set_unix_permissions "/tmp/nzbdrone_backup"
-    set_unix_permissions "/tmp/nzbdrone_update"
+    if [ -d "/tmp/nzbdrone_backup" ] || [ -d "/tmp/nzbdrone_update" ] || [ -d "/tmp/sonarr_backup" ] || [ -d "/tmp/sonarr_update" ]; then
+        set_unix_permissions "/tmp/nzbdrone_backup"
+        set_unix_permissions "/tmp/nzbdrone_update"
+        set_unix_permissions "/tmp/sonarr_backup"
+        set_unix_permissions "/tmp/sonarr_update"
```
Are these still needed? I don't see anything being created there during upgrades. FYI these paths won't work in DSM7

maxrogers replied:
It might be wise to leave these until the next update to ensure there are no permissions errors that cause any problems during the upgrade. I know that these were added to most packages previously to resolve a permissions upgrade error...

publicarray replied:
Cool, thanks for clarifying.

publicarray commented on cross/sonarr/Makefile:
```diff
@@ -1,9 +1,9 @@
-PKG_DIST_SITE = https://download.sonarr.tv/v2/master/mono
-PKG_DIR = NzbDrone
+PKG_DIST_NAME = $(PKG_NAME).main.$(PKG_VERS).linux.$(PKG_EXT)
+PKG_DIST_SITE = http://download.sonarr.tv/v3/main/$(PKG_VERS)
```
https please.

```suggestion
PKG_DIST_SITE = https://download.sonarr.tv/v3/main/$(PKG_VERS)
```

ccwillia commented:
> Yeah that's not really related to this PR/Sonarr but it looks like it's the end of the line for mono based software on 88f6281.

Thats disappointing i was looking to move to Sonarr V3 for 88f6281.  Hopefully it can be moved to .NET core at some point like Radarr

publicarray commented:
@ccwillia Sonarr V3 (.NET Core) dropped support for 88f6281, hi3535 and evansport (ARMv5, ARMv7l and i686)

Taloth commented:
> Sonarr V3 (.NET Core) dropped support for 88f6281, hi3535 and evansport (ARMv5, ARMv7l and i686)

You mean Radarr V3? :smile:

FYI. For Sonarr v3, versions below mono 5.18 show a persistent health check Notice, below 5.4 show an Health Check Error.   
Atm it still works between 5.4 and 5.18  coz I didn't bump our dependencies, but I might have to do that for v3's lifecycle hence the notice to update.   
I haven't tested below 5.4 but it's not supported anyway.

hgy59 commented on cross/sonarr/Makefile:
```diff
@@ -1,9 +1,9 @@
-PKG_DIR = NzbDrone
+PKG_DIST_NAME = $(PKG_NAME).main.$(PKG_VERS).linux.$(PKG_EXT)
+PKG_DIST_SITE = https://download.sonarr.tv/v3/main/$(PKG_VERS)
+PKG_DIR = Sonarr
```
Yes, I like the long lasted rename to Sonarr!
```suggestion
PKG_DIR = $(PKG_NAME)
```

hgy59 commented on spk/sonarr/Makefile:
```diff
@@ -1,12 +1,12 @@
 SPK_NAME = nzbdrone
```
I propose to use `Sonarr` as package name
Once we have to do the breaking change, why not now?

maxrogers replied:
When I looked into doing this previously there were some major issues that I didn't find an easy resolution for (at the time).

If I remember correctly a couple of the issues as a result of making this change were:

- By making this change synology package manager would now treat this as a new/separate package (would no longer use the standard package upgrade processes). This means that we would have to manually add a check for the old database and also build-in the database migration from the old app (via scripting). This was doable but added more potential problems than I wanted to deal with at the time.
- Synology package manager will not allow you to install a package with the same webports and a couple other items. This means that this package would require some scripting to work around that install issue. If I remember correctly you would need to have it start the install with different webports set in the new package, then remove the old app or change the webports via scripting as part of the process.
- In addition, and to potentially resolve the previous point, a bigger challenge (for me) was that I did not find a good way to add scripting to stop/uninstall the old app via scripting/commands within this new app. It seemed it would open up too many potential issues.

Perhaps there were more issues, I can't remember. However, I think this would be kind of nice to have if someone wants to make all of the above work. I don't know if another package has gone through this process before and might already have some of this scripting added to a package upgrade/rename. Another option is to make this a separate v3 package and force users to manually upgrade/migrate.

Safihre replied:
Let's please not make this complicated just for a name that a regular users wouldn't see!

maxrogers replied:
> Let's please not make this complicated just for a name that a regular users wouldn't see!

That's what we decided last time and I agree

publicarray replied:
I have to look at the documentation but in DSM7 there is a way to replace a package wich is exactly what we need to address these issues. We just need to wait until it's released

publicarray replied:
Also agree that for this PR we should leave it as is. It's not worth work on on until DSM7 is released

publicarray commented:
@hgy59 I've managed to do a smooth migration. I just need other person to test this just to make sure.

hgy59 commented:
> @hgy59 I've managed to do a smooth migration. I just need other person to test this just to make sure.

This might need a manual installation for Upgrades, as the package center knows the new package name only.
Unless we could publish a `nzbdrone` package that does the migration to sonarr. If this is possible, we only need to publish the same package as nzbdrone and sonarr once.

publicarray commented:
@hgy59 I don't follow what you're saying. From the DSM point of view the package on synocommunity is called `nzbdrone`. The last change here copies the data and installs `sonarr` and removes `nzbdrone`. Only downside is that on synocommunity.com/package-center it might look strange having 2  packages labelled `Sonarr`. 

Thanks @houndtt for testing. 

The `install configue file fail!` is somewhat expected since the same port for the firewall is used. Not sure how to fix that.

hgy59 commented:
> @hgy59 I don't follow what you're saying. From the DSM point of view the package currently on synocommunity is called `nzbdrone`. The last change here copies the data and installs `sonarr` and removes `nzbdrone`. Only downside is that on synocommunity.com/package-center it might look strange having 2 packages labelled `Sonarr`.

@publicarray as you renamed `SPK_NAME` in  `spk/sonarr/Makefile` from `nzbdrone` to `sonarr` the synocommunity package center will offer this as `sonarr` and the diskstations will not use it as update for installed `nzbdrone` packages.
So the final `nzbdrone`package that renames to `sonarr` must still have the package name `nzbdrone`.
Otherwise you cannot use the package center to do this final update but must install the spk manually.

Safihre commented:
Guys, please don't do this! Nobody of the regular users care about the package internal (!!) name. Only the few people here care.
Why would a regular user have to deal with some extra steps and potential failure points just for an insignificant name change?
Why do this? What is really the point?

hgy59 commented:
The regular user for regular use cases does not care.

**But:**
- why should you search for nzbdrone if you need the logfiles of sonarr?
- the github action dependency resolvement has some specific code for sonarr as the package is called nzbdrone
- currently I am redesigning packages that use the folder `/usr/local/{package-name}` as this is not supported on dsm7 - and here again, manual analysis is required as folder and package name do not match.

waxilian commented:
> Guys, please don't do this! Nobody of the regular users care about the package internal (!!) name. Only the few people here care.
> Why would a regular user have to deal with some extra steps and potential failure points just for an insignificant name change?
> Why do this? What is really the point?

I agree with @Safihre on this. IMO it seems DSM7 would be the time to make the change to clean-up the package via better/cleaner methods available in DSM7 to do so.

maxrogers commented:
I agree with @Safihre on this. IMO it seems DSM7 would be the time to make the change to clean-up the package via better/cleaner methods available in DSM7 to do so.

maxrogers commented:
> Guys, please don't do this! Nobody of the regular users care about the package internal (!!) name. Only the few people here care.
> Why would a regular user have to deal with some extra steps and potential failure points just for an insignificant name change?
> Why do this? What is really the point?

I agree with @Safihre on this. IMO it seems DSM7 would be the time to make the change to clean-up the package via better/cleaner methods available in DSM7 to do so.

publicarray commented:
I'm happy to revert this for now. My main reason for doing this that `nzbdrone` still pops up in one place in the UI (permissions, logs etc.) Occasionally users are confused by this.

Does anyone know a case where Synology did it with their packages? I remember ThreatPrevention moved from IntrusionPrevention and disabled IntrusionPrevention, but it also happened on user manual user action. https://archive.synology.com/download/Package

AFSIK a package update without manual installation like this hasn't been done before.

publicarray commented:
@hgy59 I think we can publish the package as is (I want to get this v3 out the door soon) and work a new PR to address the renaming. I'm very much for renaming to Sonarr just not in this PR.

dougw03 commented:
I am trying to install on Synology DS920+ and am running into the error "Please install the following package(s) before installing this package: mono (any version later than 3.6)." I tried installing v 4.6.2 v0096 https://archive.synology.com/download/Package/Mono but still getting the error. Can someone please advise? Thanks

publicarray commented:
Hi, only our mono package is supported. You need to install a mono version from us: https://synocommunity.com/package/mono

Chrisophogus commented:
Any tips for getting this version to function fully on an RTD1296 processor? 
Unable to connect to Skyhook etc. to import show info. As far as I can ascertain this is due to Mono but I can't see a way to get around this. Found a couple of threads here with fixes but they were inconclusive and from quite a while ago.

ta264 commented:
Mono 5.20 is broken on rtd1296. Apparently 5.8 might work, or you can sideload docker and work around it that way. It won't be long before minimum Sonarr mono version is 5.20 :/

Wuszek commented:
@Chrisophogus try 5.8 from [here](https://synocommunity.com/package/mono)
But if You cant use docker like myself, Sonarr will stop working sooner or later.

ta264 commented:
If you're feeling brave you can install docker with these instructions https://wiki.servarr.com/Docker_ARM_Synology

Chrisophogus commented:
Thanks both for the pointers. Rolling back to Mono 5.8 was the immediate fix. I think the long term is going to be the docker solution. Thanks again for that link, an actual easy to understand process to install Docker on this thing.

publicarray commented:
@Wuszek @houndtt @ta264 I've created a wiki entry since this has come up a lot: https://github.com/SynoCommunity/spksrc/wiki/Radarr-Sonarr-Lidarr-Jackett. Would you mind updating it with the package versions that work for you or anything else that could help users or has been repeated here elsewhere? Thanks

Wuszek commented:
@publicarray sure.
Actually I'm running rtd1296 (armv8) on DS218 and Radarr .NET core version works like a charm. The same with Jackett .NET and Lidarr .NET (i'm not using Lidarr on daily basis, just downloaded and installed for tests).

In case of Sonarr, that still requires mono in v3, (at least for now), indeed I need to use mono v5.8, because there is some bug related with https/ssl certificate in newer versions, if I can remember correctly.

GandulfM commented:
@ Christophogus:
I did the same on my DS418 a few weeks ago and it worked fine, BUT since a few days Sonarr stopped working again with Mono 5.8. It cannot contact the indexer NZBgeek anymore. Wonder if it still works for you?

timrettop commented:
@GandulfM Sonarr phantom branch latest update works fine with NZBGeek for me on DS420j with Mono 5.8. I just have the general error that 5.8 is not supported.

bakerboy448 commented:
Sonarr phantom is dead and out of date. Two main versions have been released since phantom was killed

timrettop commented:
Your right, i'm on main, 3.0.6.1196


On Sat, Apr 10, 2021 at 6:21 PM bakerboy448 ***@***.***>
wrote:

> Sonarr phantom is dead and out of date. Two main versions have been
> released since phantom was killed
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynoCommunity/spksrc/pull/3803#issuecomment-817230929>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADKFYM2UTXOUXURLRFFWL3TID2TLANCNFSM4JMJAV5Q>
> .
>

timrettop commented:
Oh but I've been updating via the Sonarr UI, haven't updated via DSM in a while.

tvdbergh commented:
I've been trying to update to Sonarr v3 but just noticed the new package doesn't support the evansport architecture of my DS214Play. Just wondering if my NAS is too old and not supported anymore? (Radarr v3 does does still seem to be supported)

GandulfM commented:
This is what i just did for my DS418:
Here you find the architecture of your Synology: https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model
Then find the Mono version here: https://synocommunity.com/package/mono
And for Sonarr: https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build
I took the mono 5.8 and found  the Sonarr version is not the newest, but installed anyway. However get warnings to update, but Sonarr is running now!

origami-army commented:
bit stuck with the new update to Sonarr V3 on my DS218play - RTD1296 processor.
Had Sonarr v2 with mono 5.8 and seemed to work with indexers (did encounter the bug with 5.20 few months back and had to revert)
 - Radarr updated to V3 and seemed to work well with mono 5.8
 - Wanted to try Sonarr V3 and updated - indexers stopped working unfortunately - thought it was related to mono 5.8 so updated to 5.20 but now still stuck with errors connecting:
 
[v3.0.5.1144] System.Net.WebException: Error: TrustFailure (One or more errors occurred.): 'https://drunkenslug.com/api?t=caps&apikey=(removed) ---> System.Net.WebException: Error: TrustFailure (One or more errors occurred.) ---> System.AggregateException: One or more errors occurred. ---> System.Security.Authentication.AuthenticationException: A call to SSPI failed, see inner exception. ---> 

anyone have any ideas or dealing with similar issue?

Chrisophogus commented:
That's the issue I had.
Revert back to 5.8 by downloading the package directly from
https://synocommunity.com/package/mono

Sonarr is complaining it's on 5.8 but it's working fine.


On Wed, 14 Apr 2021 at 09:41, origami-army ***@***.***> wrote:

> bit stuck with the new update to Sonarr V3 on my DS218play - RTD1296
> processor.
> Had Sonarr v2 with mono 5.8 and seemed to work with indexers (did
> encounter the bug with 5.20 few months back and had to revert)
>
>    - Radarr updated to V3 and seemed to work well with mono 5.8
>    - Wanted to try Sonarr V3 and updated - indexers stopped working
>    unfortunately - thought it was related to mono 5.8 so updated to 5.20 but
>    now still stuck with errors connecting:
>
> [v3.0.5.1144] System.Net.WebException: Error: TrustFailure (One or more
> errors occurred.): 'https://drunkenslug.com/api?t=caps&apikey=(removed)
> ---> System.Net.WebException: Error: TrustFailure (One or more errors
> occurred.) ---> System.AggregateException: One or more errors occurred.
> ---> System.Security.Authentication.AuthenticationException: A call to SSPI
> failed, see inner exception. --->
>
> anyone have any ideas or dealing with similar issue?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynoCommunity/spksrc/pull/3803#issuecomment-819344463>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIJKXMG7UPCWR56PTKXCX3TIVIMZANCNFSM4JMJAV5Q>
> .
>

origami-army commented:
Hmm - was having issue with V3 & 5.8 mono prior to updating but I'll try reverting again. Do you know of a way of doing this without uninstalling Sonarr?

wkobiela commented:
Download last backup from Sonarr (zip package by ssh if you can't download it from web Sonarr interface), reinstall packages and upload backup

Wuszek commented:
Download last backup from Sonarr (zip package by ssh if you can't download it from web Sonarr interface), reinstall packages and upload backup

origami-army commented:
Wow that was unexpected - thanks for the help, resolved now. Reinstalling with mono 5.8 (even though I was originally on 5.8 when it first stopped working) allowed indexers to work again. Restore now doesn't require SSH to upload so was able to do it seamlessly from the Sonarr settings

GandulfM commented:
Hi guys,
Anybody having problems with Sonarr on their Synology NAS I advice to follow up on my post of 3 days ago.
The various architectures of Synology NASes make that not all NASes can work with the Mono and Sonarr packages advised by Package Center, like my DS418 with rtd1296 (armv8) architecture.
So, if problems with Sonarr, first:

- Find the architecture of your Synology: https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model

- Then find the Mono version here: https://synocommunity.com/package/mono

I advice to download the 5.8 version to your computer and then manually install to your Synology.

- Then find the Sonarr version you need for your NAS architecture here:  https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build

And download to your computer and manually install this version.

- Ignore advises to update Mono and Sonarr to latest versions.

I understand from Fora, they are working on a version of Sonarr that doesn't need Mono to run, like already is the case with Radarr. 
Probably end of this year with DSM v7?
Good luck all!

publicarray commented:
Thanks for the information  @GandulfM. We do have a wiki entry, but you are welcome to add to it: https://github.com/SynoCommunity/spksrc/wiki/Radarr-Sonarr-Lidarr-Jackett

GandulfM commented:
> 
> 
> Thanks for the information @GandulfM. We do have a wiki entry, and you are welcome to add to it: https://github.com/SynoCommunity/spksrc/wiki/Radarr-Sonarr-Lidarr-Jackett

@publicarray, for my DS418 I manually installed https://github.com/Safihre/spksrc/releases/download/sonarr-v3-build/nzbdrone_rtd1296-6.1_20201116-19.spk
It worked fine for 5 days, but today, when I open Sonarr I get a blank page with:
Object reference not set to an instance of an object
Version 3.0.3.750
So Sonarr stopped again.
Of course i have mono 5.8 installed too.

However, after a restart of my diskstation Sonarr was running again!

Segelfeldt commented:
I get the problem that Sonarr and radar says
```
{
  "message": "NotFound"
}
```

The services is stopped and when i try to start it its gets backed to stopped state.

Tried to follow @GandulfM steps above but when I am trying to roll back but when i manually install Mono it says current package version is 5.20.1.34-16. You cannot roll back to the older version 5.8.0.108-11. I am unable to uninstall Mono since it says Sonarr and Radarr is dependent on it. How do i solve this?

I noticed some say i can download the backup file, i were considering doing that and moving do Docker but were can i find the backup files for Sonarr and Radarr?

GandulfM commented:
What you need to do is uninstalling Mono, Sonarr and Radarr.
Then manually install Mono 5.8, Sonarr for your architecture and Radarr, as described above https://github.com/SynoCommunity/spksrc/pull/3803#issuecomment-819551856

Segelfeldt commented:
> What you need to do is uninstalling Mono, Sonarr and Radarr.
> Then manually install Mono 5.8, Sonarr for your architecture and Radarr, as described above [#3803 (comment)](https://github.com/SynoCommunity/spksrc/pull/3803#issuecomment-819551856)

Do i need to take a backup first? If so how? or will the data remain on my drives even if i uninstall and when i reinstall they will use the data that is already there?

GandulfM commented:
@Segelfeldt 
It's always been advised to make a backup, but I didn't; what's the use of a backup that doesn't work?
However, you may want to copy the api key from your indexer and a list of wanted videos, like I did.
Also, the latest version of Radarr doesn't use Mono, so possibly an upgrade suffices and just uninstall Mono and Sonarr.
My DS418 now runs perfectly for a month with Mono 5.8 and appropriate Sonarr version for it's architecture. For Radarr i have the latest version from Package Center.
Good luck!

Safihre commented:
I'm locking this issue, this is not the place (anymore) for instructions how to get it running.
